shared/js/ui/models/whitelist.es6.js

```diff
@@ -22,6 +22,10 @@ Whitelist.prototype = window.$.extend({},
         value: false
       }
       })
+
+      // Update list
+      // use splice() so it reindexes the array
+      this.list.splice(itemIndex, 1)
```

Comment:
To expand on the comment, if I used 'delete list[itemIndex]' it would have been fine too, but it wouldn't have re-indexed the list.
Basically if the list was [google.com, msnbc.com]:
- After 'delete list[1]', it would be  [google.com, undefined]
- After 'this.list.splice(1, 1)', it's [google.com]

You can test the difference by console.logging the result here and then replacing splice() with delete